RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc

```diff
@@ -103,15 +106,17 @@ MuonME0DetLayerGeometryBuilder::buildLayer(int endcap,
     }
 
   }
-  LogTrace(metname) << "About to make a MuRingForwardLayer";
-  result = new MuRingForwardLayer(frontRings);
-  
-  LogTrace(metname) << "New MuRingForwardLayer with " << frontRings.size()
-                    << " and " << backRings.size()
-                    << " rings, at Z " << result->position().z()
-                    << " R1: " << result->specificSurface().innerRadius()
-                    << " R2: " << result->specificSurface().outerRadius();
   
+  LogTrace(metname) << "About to make a MuRingForwardLayer";
+  if(frontRings.size()!=0) result = new MuRingForwardLayer(frontRings);
```

Comment:
@jshlee: Yes, it would be good to fix this issue. Changing to pointers to std::shared_ptr or std::unique_ptr (as appropriate) would mean the deletion would happen automatically.